spinnaker-monitoring-daemon/spinnaker-monitoring/stackdriver_service.py

```diff
@@ -243,7 +248,55 @@ def project_to_resource(self, project):
       raise ValueError('No project specified')
     return 'projects/' + project
 
+  def __auto_audit_metric_descriptors(self, options):
+    """The janitor function attempts to bring Stackdriver into compliance.
+
+    If the metric descriptors are already as expected then we'll disable
+    the janitor for the rest of the process' lifetime. Otherwise we'll
+    continue to call it and try again around every JANITOR_PERIOD seconds
+    to give time for the system to settle down.
+
+    The reason we expect to have problems is that old replicas are still
+    running and recreating the descriptors we are trying to delete when
+    stackdriver automatically creates metrics they are attempting to write.
+    If this is the case, we'll keep trying to clear them out until, eventually,
+    the old processes are no longer around to overwrite us.
+
+    Should something re-emerge then we'll be messed up until the next restart.
+    Note that each replica of each service is probably trying to create all
+    the descriptors so there is a lot of activity here. Since the descriptors
+    are all the same, there should not be a problem with these replicas
+    conflicting or needing coordination.
+
+    Note if management is disabled then this will be in a stable state
+    though still inconsistent with stackdriver because there will not
+    be any errors or activity performed.
+    """
+    secs_remaining = self.__next_janitor_time - time.time()
+    close_enough = 10
```

Comment:
Is there a reason we can't just wait until the next scheduled time, or make the `JANITOR_PERIOD` 10 seconds smaller?  Keeping track of a `close_enough` value seems to add (minor) complexity without any real benefit.

Reply:
ok